sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScan.scala

```diff
@@ -50,7 +50,10 @@ case class ParquetScan(
   override def isSplitable(path: Path): Boolean = {
     // If aggregate is pushed down, only the file footer will be read once,
     // so file should not be split across multiple tasks.
-    pushedAggregate.isEmpty
+    pushedAggregate.isEmpty &&
+    // SPARK-39634: Allow file splitting in combination with row index generation once
```

Comment:
nit: looks like the comment is off. I guess it is for L56.

Reply:
I assume it needed an indent.